executor/runtime/docker/docker.go

```diff
@@ -147,6 +147,10 @@ func NewDockerRuntime(ctx context.Context, m metrics.Reporter, dockerCfg Config,
 		return nil, err
 	}
 
+	// We bind-mount tini in as /sbin/docker-init to ensure we can always
+	// depend on it being there, regardless of the host docker configuration.
+	defaultBindMounts := []string{dockerCfg.tiniPath + ":/sbin/docker-init:ro"}
```

Comment:
This should be mounted at `/dev/init`, no? Do you plan to still use docker to do the entry point rewriting? If you do, that is equivalent of forcing the flag on all the time, which might be easier?

Reply:
Older docker version use `/dev/init`, but newer ones (my laptop / devstack) use the new path.
It won't hurt on older versions.

I don't plan on doing the rewriting, and we don't need to with the `--init` flag (it does the rewrite for us).

Reply:
Cool, this commit description confused me.

```
To do this, I bind mount tini from the host, instead of depending
on --init being set on the docker daemon. This works even if it
is set.```

Reply:
Oh, yea I guess there are *two* flags here we are confusing.

One is `--init-path` on the docker daemon. This PR is to make it so we don't depend on that.

The second is `--init` on docker run. This bind mounts in (something) and rewrites entrypoint. We *do* still need this. Just with my PR, my bind mount wins and get used, regardless of `--init-path`.